controller/cmd/tap/main.go

```diff
@@ -43,17 +43,17 @@ func Main(args []string) {
 	k8sAPI, err := k8s.InitializeAPI(
 		*kubeConfigPath,
 		true,
-		k8s.CJ,
-		k8s.DS,
-		k8s.SS,
-		k8s.Deploy,
-		k8s.Job,
-		k8s.NS,
-		k8s.Pod,
-		k8s.RC,
-		k8s.Svc,
-		k8s.RS,
-		k8s.Node,
+		k8s.RT(k8s.CJ),
+		k8s.RT(k8s.DS),
+		k8s.RT(k8s.SS),
+		k8s.RT(k8s.Deploy),
+		k8s.RT(k8s.Job),
+		k8s.RT(k8s.NS),
+		k8s.RT(k8s.Pod),
+		k8s.RT(k8s.RC),
+		k8s.RT(k8s.Svc),
+		k8s.RT(k8s.RS),
+		k8s.RT(k8s.Node),
```

Comment:
Having to modify all of the constructors in this way is surprising. What motivates this? Can we instead do this transformation within `InitializeAPI`. If this really is something that needs to be called in all of these contexts, the `RT` abbreviation means nothing to me... is this something that should be understood in the broader k8s ecosystem? (The other abbreviations are canonical within k8s afaiu).

Reply:
The reason for this is that instead of simply an enum that is encoded as an int, I want to pass a struct that contains the resource type and an optional namespace scope. The problem is that  cannot really have enum constants in go and this necessitates this constructor call here... If that wasnt the case the change would be less visible. We can make it so this is somehow handled within the `InitializeAPI` method but we would still need to pass aditional info somehow. So all in all I do not really see a good way to avoid that. RT should stand for ResourceType but clearly that is poor choice of abbreviation. I can split the change in two is that helps

Reply:
How about having `initializeAPI()` build that `APIResource` using `corev1.NamespaceAll`, leaving the calling sites for all the old controllers untouched, and have a new `InitializeNamespacedAPI()` only used by the service-mirror that would receive the namespace as an argument and build `APIResource` with that namespace?

Reply:
I was thinking abou that but in our particular case we have some resources that are namespaces (secrets) and some that are not that want to share the same api. So thi will not work as we would want to specify namespace constraints per resource in this case. I say we just do it the go way and copy the code all over ;)

Reply:
Okey... Now that I made the change it is clear that we need to touch quite a bit less stuff than originally to make this all work. So it is worth it. Now client.go is almost untouched.  Good suggestion !